modules/git/object_format.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package git
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type ObjectFormatID int
+
+const (
+	Sha1 ObjectFormatID = iota
```

Comment:
Why it needs "integer" number?

I think `type ObjectFormatID string` and `Sha1 = "sha1"` work better?

Reply:
I think so.

Reply:
There is another function that does this from `string` already, at the bottom of the file.

`git.ObjectFormatFromString()`

Basically, there is no need for this to be int or string in particular, but I wanted to move at least temporarily away from string to catch  errors like passing "SHA1" vs. "sha1".

Finally, maybe my prejudice shows, coming from C and C++ world shows, where enums are ints :smile:

Reply:
#28469